base/compiler/abstractinterpretation.jl

```diff
@@ -567,21 +565,30 @@ function abstract_iteration(interp::AbstractInterpreter, @nospecialize(itft), @n
     stateordonet === Bottom && return Any[Bottom]
     valtype = statetype = Bottom
     ret = Any[]
-    stateordonet = widenconst(stateordonet)
-    while !(Nothing <: stateordonet) && length(ret) < InferenceParams(interp).MAX_TUPLE_SPLAT
-        if !isa(stateordonet, DataType) || !(stateordonet <: Tuple) || isvatuple(stateordonet) || length(stateordonet.parameters) != 2
+    # Try to unroll the iteration up to MAX_TUPLE_SPLAT, which covers any finite
+    # length iterators, or interesting prefix
+    while true
+        stateordonet_widened = widenconst(stateordonet)
```

Comment:
```suggestion
        stateordonet_widened = widenconst(stateordonet)
        if stateordonet_widened === Nothing
            return ret
        end  
```
So this test can be skipped below.
(Is there a way to suggest an addition only?)